python/ray/ml/tests/test_xgboost_predictor.py

```diff
@@ -0,0 +1,70 @@
+import os
+from ray.ml.predictors.integrations.xgboost import XGBoostPredictor
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.constants import MODEL_KEY, PREPROCESSOR_KEY
+import json
+
+import numpy as np
+import xgboost as xgb
+import tempfile
+
+
+class DummyPreprocessor(Preprocessor):
+    def transform_batch(self, df):
+        return df * 2
+
+
+dummy_data = np.array([[1, 2], [3, 4], [5, 6]])
+dummy_target = np.array([0, 1, 0])
+
+
+predictor_cls = XGBoostPredictor
+model = xgb.XGBClassifier(n_estimators=10).fit(dummy_data, dummy_target).get_booster()
+
+
+def get_num_trees(booster: xgb.Booster) -> int:
+    data = [json.loads(d) for d in booster.get_dump(dump_format="json")]
+    return len(data) // 4
+
+
+def test_init():
+    preprocessor = DummyPreprocessor()
+    preprocessor.attr = 1
+    predictor = predictor_cls(model=model, preprocessor=preprocessor)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # This somewhat convoluted procedure is the same as in the
+        # Trainers. The reason for saving model to disk instead
+        # of directly to the dict as bytes is due to all callbacks
+        # following save to disk logic. GBDT models are small
+        # enough that IO should not be an issue.
+        model.save_model(os.path.join(tmpdir, MODEL_KEY))
+        checkpoint = Checkpoint.from_dict({PREPROCESSOR_KEY: preprocessor})
+        checkpoint.to_directory(path=tmpdir)
+
+        checkpoint = Checkpoint.from_directory(tmpdir)
+        checkpoint_predictor = predictor_cls.from_checkpoint(checkpoint)
+
+    assert get_num_trees(checkpoint_predictor.model) == get_num_trees(predictor.model)
+    assert checkpoint_predictor.preprocessor.attr == predictor.preprocessor.attr
+
+
+def test_predict():
+    preprocessor = DummyPreprocessor()
+    predictor = predictor_cls(model=model, preprocessor=preprocessor)
```

Comment:
Is there also a way to test that the preprocessor is being executed?

Reply:
Added a mock to the preprocessor itself